users/arxiv/users/auth/__init__.py

```diff
@@ -131,18 +127,13 @@ def load_session(self) -> Optional[Response]:
         if isinstance(session, Exception):
             logger.debug('Middleware passed an exception: %s', session)
             raise session
-
-        # If the legacy database is available, we should use it to authorize
-        # the request.
+        # use legacy DB to authorize request if available
         if legacy.is_configured():
-            logger.debug('No session; attempting to get legacy session')
-            response = self.detect_and_clobber_dupe_cookies()
-
-            # Return that is anything other than None is treated as a response;
-            # request handling stops here.
-            if response is not None:
-                return response
-            session = self._get_legacy_session()
+            logger.debug('No session; attempting to get legacy from cookies')
+            session = next(filter(bool,
```

Comment:
Neat! Some brief documentation would be helpful here for future readers to unpack this.

Reply:
I moved that code out to a method.